watertap/property_models/SOP_prop_pack.py

```diff
@@ -0,0 +1,551 @@
+            mutable=True,
+            initialize=1e3,
+            units=units.kg * units.m**-3,
+            doc="Component mass density",
+        )
+
+        self.dens_mass_phase_comp["Liq", "H2O"] = 1e3
+        self.dens_mass_phase_comp["Liq", "oil"] = 780
+
+    def _mass_frac_phase_comp(self):
+        self.mass_frac_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=0.1,
+            bounds=(1e-15, None),
+            units=units.dimensionless,
+            doc="Mass fraction",
+        )
+
+        def rule_mass_frac_phase_comp(b, p, j):
+            return (
+                b.mass_frac_phase_comp[p, j] * b.flow_mass_phase[p]
+                == b.flow_mass_phase_comp[p, j]
+            )
+
+        self.eq_mass_frac_phase_comp = Constraint(
+            self.params.phase_list,
+            self.params.component_list,
+            rule=rule_mass_frac_phase_comp,
+        )
+
+    def _flow_vol_phase_comp(self):
+        self.flow_vol_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=1e-6,
+            domain=NonNegativeReals,
+            units=units.m**3 / units.s,
+            doc="Component volumetric flowrate",
+        )
+
+        def rule_flow_vol_phase_comp(b, p, j):
+            return (
+                b.flow_vol_phase_comp[p, j] * b.dens_mass_phase_comp[p, j]
+                == b.flow_mass_phase_comp[p, j]
+            )
+
+        self.eq_flow_vol_phase_comp = Constraint(
+            self.params.phase_list,
+            self.params.component_list,
+            rule=rule_flow_vol_phase_comp,
+        )
+
+    def _flow_vol_phase(self):
+        self.flow_vol_phase = Var(
+            self.params.phase_list,
+            initialize=1e-3,
+            bounds=(None, None),
+            units=units.m**3 / units.s,
+            doc="Volumetric flow rate",
+        )
+
+        def rule_flow_vol_phase(b, p):
+            return b.flow_vol_phase[p] == sum(
+                b.flow_vol_phase_comp[p, j] for j in b.params.component_list
+            )
+
+        self.eq_flow_vol_phase = Constraint(
+            self.params.phase_list, rule=rule_flow_vol_phase
+        )
+
+    def _flow_mass_phase(self):
+        self.flow_mass_phase = Var(
+            self.params.phase_list,
+            initialize=1e-3,
+            bounds=(None, None),
+            units=units.kg / units.s,
+            doc="Mass flow rate",
+        )
+
+        def rule_flow_mass_phase(b, p):
+            return b.flow_mass_phase[p] == sum(
+                b.flow_mass_phase_comp[p, j] for j in b.params.component_list
+            )
+
+        self.eq_flow_mass_phase = Constraint(
+            self.params.phase_list, rule=rule_flow_mass_phase
+        )
+
+    def _dens_mass_phase(self):
+        self.dens_mass_phase = Var(
+            self.params.phase_list,
+            initialize=1e3,
+            bounds=(1e2, 2e3),
```

Comment:
not sure, but it could be a safer bet to relax this upper bound

Reply:
@adam-a-a can you explain what you mean by "safer" in this context? And by "relax", do you mean to just make the upper bound higher, or to remove it altogether?